examples/index.md

```diff
@@ -162,3 +162,249 @@ Content-Type: application/vnd.api+json
   }
 }
 ```
+
+## Error Objects
+
+Examples of how [error objects](http://jsonapi.org/format/#error-objects) work.
+
+### Responses to failed resource creation:
+
+Response with one error object, with attribute error:
+
+```http
+HTTP/1.1 422 Unprocessable Entity
+Content-Type: application/vnd.api+json
+
+{
+  "errors": [
+    {
+      "title":  "Invalid Attribute",
+      "source": { "pointer": "data/attributes/first-name" },
+      "detail": "First name must contain at least three characters."
+    }
+  ]
+}
+```
+
+Notice that the attribute `first-name` is dasherized.  Dasherized attributes are preferred
+over underscored.
+
+Notice also that the "detail" is specific to this occurrence of the problem,
+whereas the "title" is more generic.
+
+Notice that the "source" points to the associated attribute with the error.
+
+Response with one error object, with relationship error:
+
+```http
+HTTP/1.1 422 Unprocessable Entity
+Content-Type: application/vnd.api+json
+
+{
+  "errors": [
+    {
+      "title":  "Missing Association",
+      "source": { "pointer": "data/relationships/author" },
+      "detail": "Post author cannot be blank."
+    }
+  ]
+}
+```
+
+Notice that the "source" points to the associated relationship with the error.
+
+Response with one error object, with parameter error:
+
+```http
+HTTP/1.1 400 Bad Request
+Content-Type: application/vnd.api+json
+
+{
+  "errors": [
+    {
+      "title":  "Missing Query Parameter",
+      "source": { "parameter": "data" },
+      "detail": "The required parameter, data, is missing."
```

Comment:
@ethanresnick Thanks for confirming my suspicions.  I glanced through some docs on the web and didn't immediately find anything that distinguished url params from body params when discussing query params.

And I suppose it would be up the implementation to decide whether to ignore an invalid or missing parameter or to return an error.

I think I can keep this example as a POST with body `{ "datum": { } }` to return an error object with `"source": { "pointer": "/data" }`  and add a new one for parameter, but I can't think of an  example where it wouldn't make sense for the client to ignore an invalid or missing parameter.

`GET http://www.example.com/api/users/1/secrets.json`  could return a `missing auth_key parameter`?